apps/docs/content/guides/realtime/authorization.mdx

```diff
@@ -0,0 +1,293 @@
+
+Create a policy for `select` on `realtime.messages` where `realtime.messages.extension` is `presence`.
+
+```sql
+create policy "authenticated users can listen to presence changes in a room"
+on "realtime"."messages"
+for select
+to authenticated
+using (
+  exists (
+    select
+      ru.user_id
+    from
+      rooms_users ru
+    where
+      ru.user_id = (select auth.uid())
+      and ru.topic = (select realtime.topic())
+      and realtime.messages.extension in ('presence')
+  )
+);
+```
+
+#### Allow users to send Presence messages on a channel
+
+To update the Presence status for a user create a policy for `insert` on `realtime.messages` where the value of `realtime.messages.extension` is `presence`.
+
+```sql
+create policy "authenticated users can register and update their presence in a room"
+on "realtime"."messages"
+for insert
+to authenticated
+with check (
+  exists (
+    select
+      ru.user_id
+    from
+      rooms_users ru
+    where
+      ru.user_id = (select auth.uid())
+      and ru.name = (select realtime.topic())
+      and realtime.messages.extension in ('presence')
+  )
+);
+```
+
+### Presence and Broadcast
+
+Authorize both Presence and Broadcast by including both extensions in the `where` filter.
+
+#### Broadcast and Presence read
+
+Authorize Presence and Broadcast read in one RLS policy.
+
+```sql
+create policy "authenticated user can listen to Broadcast and Presence messages in room"
+on "realtime"."messages"
+for select
+to authenticated
+using (
+  exists (
+    select
+      ru.user_id
+    from
+      rooms_users ru
+    where
+      ru.user_id = (select auth.uid())
+      and ru.topic = (select realtime.topic())
+      and realtime.messages.extension in ('broadcast', 'presence')
+  )
+);
+```
+
+#### Broadcast and Presence write
+
+Authorize Presence and Broadcast write in one RLS policy.
+
+```sql
+create policy "authenticated user can send Broadcast and Presence messages in a room"
+on "realtime"."messages"
+for insert
+to authenticated
+with check (
+  exists (
+    select
+      ru.user_id
+    from
+      rooms_users ru
+    where
+      ru.user_id = (select auth.uid())
+      and ru.name = (select realtime.topic())
+      and realtime.messages.extension in ('broadcast', 'presence')
+  )
+);
+```
+
+## Interaction with Postgres Changes
+
+Realtime Postgres Changes are separate from Channel authorization. The `private` Channel option does not apply to Postgres Changes.
+
+When using Postgres Changes with RLS, database records are sent only to clients who are allowed to read them based on your RLS policies.
+
+## Updating RLS policies
+
+When you change an RLS policy these changes are picked up when a user connects to a Channel.
+
+When a user connects, Realtime queries your database and generates a representation of the permissions for a user for a Channel.
+
+Permissions are cached for the duration of the connection so we don't have to query your database for each Channel message.
```

Comment:
Not the duration of the connection but as long as client is subscribed to channel.

Reply:
A client can "leave" a topic and not disconnect but when the server kills the Channel I'm pretty sure the connection is actually terminated. eg when one of the `RealtimeChannel` callbacks respond with `{:stop, :shutdown, socket}` the process is stopped.